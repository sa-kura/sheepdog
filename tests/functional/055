#!/bin/bash

# Test md with single failure, multiple failure and all failure

. ./common

MD=true

for i in `seq 0 6`; do
	_start_sheep $i
done
_wait_for_sheep 7
_cluster_format -c 6
_vdi_create test 120M -P

# simulate one disk failure
_safe_remove $STORE/0/d1
_random | $DOG vdi write test
_wait_for_sheep_recovery 0
$DOG vdi check test
$DOG cluster info | _filter_cluster_info

# simulate multiple disk failure
_safe_remove $STORE/1/d1
_random | $DOG vdi write test &
_wait_for_sheep_recovery 0
_safe_remove $STORE/1/d2
_wait_for_sheep_recovery 0
wait
$DOG vdi check test | sort
$DOG cluster info | _filter_cluster_info

# simulate all disks failure
_safe_remove $STORE/1/d0
_random | $DOG vdi write test
_wait_for_sheep_recovery 0
$DOG vdi check test
$DOG cluster info | _filter_cluster_info

# simulate simultaneous multiple disks failure
_safe_remove $STORE/2/d1
_safe_remove $STORE/2/d2
dd if=/dev/zero | $DOG vdi write test
$DOG vdi check test
$DOG cluster info | _filter_cluster_info
