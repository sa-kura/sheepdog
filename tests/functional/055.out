QA output created by 055
using backend plain store
finish check&repair test
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128, 127.0.0.1:7006:128]
Response's result: Network error between sheep
failed to repair object 007c2b2500000003 from 127.0.0.1:7006 to 127.0.0.1:7001
finish check&repair test
fixed missing 7c2b2500000003
fixed missing 7c2b2500000004
fixed missing 7c2b2500000006
fixed missing 7c2b250000000d
fixed missing 7c2b250000000e
fixed missing 7c2b2500000011
fixed missing 7c2b2500000014
fixed missing 7c2b2500000016
fixed missing 7c2b2500000017
fixed missing 7c2b2500000019
fixed missing 7c2b250000001a
fixed missing 7c2b250000001b
fixed missing 7c2b250000001c
fixed missing 807c2b2500000000
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128, 127.0.0.1:7006:128]
finish check&repair test
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128, 127.0.0.1:7006:128]
finish check&repair test
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128, 127.0.0.1:7006:128]
